amethyst_derive/tests/test.rs

```diff
@@ -75,3 +75,20 @@ pub struct Outer {
     #[prefab(Component)]
     external: External,
 }
+
+#[derive(PrefabData, Clone)]
+pub struct OuterTuple(#[prefab(Component)] External);
+
+#[derive(PrefabData, Clone)]
+pub enum EnumPrefab {
+    One {
+        number: Stuff<usize>,
+    },
+    Two {
+        #[prefab(Component)]
+        component: External,
+    },
+    Three {},
+    Four,
+    Five(Stuff<usize>, #[prefab(Component)] External),
+}
```

Comment:
Could you add some actual tests (functions tagged with `#[test]`)? I think `EnumPrefab` covers the cases pretty well, we just need to check that it correctly instantiates the prefab.

Reply:
Yeah, that's a good idea.